src/Mutator/NodeMutationGenerator.php

```diff
@@ -86,27 +85,21 @@ public function __construct(
     /**
      * @return Mutation[]
      */
-    public function generate(Node $node): array
+    public function generate(Node $node): iterable
     {
-        return array_reduce(
-            $this->mutators,
-            function (array $mutations, Mutator $mutator) use ($node): array {
-                return $this->generateForMutator($node, $mutator, $mutations);
-            },
-            []
-        );
+        yield from take($this->mutators)->map(function (Mutator $mutator) use ($node) {
+            yield from $this->generateForMutator($node, $mutator);
+        });
```

Comment:
Technically we can return the pipeline here as it is, as it corresponds to the `iterable` typehint, but I want to avoid exposing this implementation detail.